projects/plugins/jetpack/tests/e2e/specs/forms/submission.test.js

```diff
@@ -0,0 +1,172 @@
+import { Plans, prerequisitesBuilder } from '_jetpack-e2e-commons/env/index.js';
+import { expect, test } from '_jetpack-e2e-commons/fixtures/base-test.js';
+import { BlockEditorPage } from '_jetpack-e2e-commons/pages/wp-admin/index.js';
+import playwrightConfig from '../../playwright.config.mjs';
+
+test.beforeAll( async ( { browser } ) => {
+	const page = await browser.newPage( playwrightConfig.use );
+	await prerequisitesBuilder( page )
+		.withCleanEnv()
+		.withLoggedIn( true )
+		.withWpComLoggedIn( true )
+		.withConnection( true )
+		.withPlan( Plans.Free )
+		.build();
+	await page.close();
+} );
+
+test.afterEach( async ( { requestUtils } ) => {
+	// List all feedback submissions.
+	// https://developer.wordpress.org/rest-api/reference/posts/#list-posts
+	const feedbackSubmissions = await requestUtils.rest( {
+		path: '/wp/v2/feedback',
+		params: {
+			per_page: 100,
+			// All possible statuses.
+			status: 'publish,future,draft,pending,private,trash',
+		},
+	} );
+
+	// Delete all feedback submissions one by one.
+	// https://developer.wordpress.org/rest-api/reference/posts/#delete-a-post
+	// "/wp/v2/feedback" does not yet support batch requests.
+	await Promise.all(
+		feedbackSubmissions.map( feedback =>
+			requestUtils.rest( {
+				method: 'DELETE',
+				path: `/wp/v2/feedback/${ feedback.id }`,
+				params: {
+					force: true,
+				},
+			} )
+		)
+	);
+} );
+
+test.describe( 'Forms: Submission', () => {
+	test( 'Submits a simple contact form', async ( { editor, page } ) => {
+		const formTitle = 'E2E Test Form';
+		await test.step( 'Visit the block editor and insert a form', async () => {
+			const blockEditor = await BlockEditorPage.visit( page );
+			await page.waitForURL( '**/post-new.php' );
+			await blockEditor.waitForEditor();
+			await blockEditor.closeWelcomeGuide();
+			await editor.insertBlock( {
+				name: 'jetpack/contact-form',
+				attributes: { formTitle },
+			} );
+
+			// TODO: Ideally it'd be possible to use `editor.canvas` from the `@wordpress/e2e-test-utils-playwright` package.
+			// This doesn't work for a non-iframed canvas - see https://github.com/WordPress/gutenberg/issues/69840.
+			const canvas = blockEditor.canvasPage.canvas();
+			const formBlock = canvas.getByRole( 'document', { name: 'Block: Form' } );
+			await formBlock.getByRole( 'button', { name: 'Add a contact form to your page.' } ).click();
```

Comment:
You can use something [like this](https://github.com/WordPress/gutenberg/blob/e49849c95d115d744ad314a255c2be5394238b82/test/e2e/specs/editor/blocks/avatar.spec.js#L40-L42):

```ts
const formBlock = editor.canvas.locator(
    'role=document[name="Block: Form"i]'
);
```

Reply:
Ah OK, it won't work if the editor is not iframed. Nevermind

Reply:
I mentioned it in the comment above the code, but `editor.canvas` doesn't work when the canvas is not iframed. I've made a gutenberg issue for the problem.

I'm guessing some of the jetpack blocks are causing the canvas not to be iframed. 🤔

Reply:
> I'm guessing some of the jetpack blocks are causing the canvas not to be iframed. 

I thought this PR took care rest of block.json registrations for blocks and iframe should now work, but I don't know details:

- https://github.com/Automattic/jetpack/pull/42929

And I assume we _would not_ need this for forms & editor iframe:

- https://github.com/Automattic/jetpack/pull/41568

There are some block.json related warnings but I don't think those would stop iframe from working:

- https://github.com/Automattic/jetpack/issues/42689

cc @jeherve @coder-karen in case you know more.

Reply:
The editor canvas is iframed now, yes, since the API version 3 PR you mentioned was merged.

You can confirm this by checking for the iframe in the page source in trunk vs when checking a PR out that hasn't been updated since all blocks were converted to v3 (eg #42852 at the time of writing).

Those warnings don't prevent the iframe from working, but instead are showing because it is loading (so we need to improve how we are enqueuing our assets in some cases).

I'm not familiar with E2E tests, but could the test environment be including other plugins that include blocks that are not using v3?

Reply:
The PR probably needs a rebase. I think I wrote some of this before that PR was merged.

Then hopefully we can use `editor.canvas` in the tests (which explicitly selects elements from the iframe canvas).

The one drawback would be that if for any reason the non-iframed canvas becomes active again tests using `editor.canvas` will start failing.

We could add an end to end test to check for the iframed canvas though, and that might provide a more obvious indication of the failure.

Reply:
Just dusting off this PR again. The tests passed on my latest run, so I think this is good now!